compiledb/__init__.py

```diff
@@ -37,16 +41,24 @@ def generate_json_compdb(instream=None, proj_dir=os.getcwd(), verbose=False, exc
 
 def write_json_compdb(compdb, compdb_path='compile_commands.json', verbose=False,
                       force=False, pretty_output=True):
-    print("## Writing compilation database with {} entries to {}".format(
-        len(compdb), compdb_path))
+    if not __is_stdout(compdb_path):
```

Comment:
Added separate logic for stdout. This turned out to be a lot less code than doing something like this [Stack Overflow article](https://stackoverflow.com/questions/17602878/how-to-handle-both-with-open-and-sys-stdout-nicely)